### PR TITLE
Add pynq-z2 board files

### PR DESCRIPTION
**Pull request by davidmallasen**
Hardcode the path to the pynq-z2 board files to avoid driver issues. However, even if the hello world still works on the FPGA, there are a few warnings arising when synthesizing. I'm marking the pull request as draft until we can figure out what these warnings are.

### CODE REVIEW

**Review by davideschiavone — CHANGES_REQUESTED** [review]
can you also pls remove from the vendor folder the pynq image?

File: hw/fpga/board_files/vendor/esl_epfl_pynq_z2_board_files.lock.hjson [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Copyright lowRISC contributors.
```
[the remark this answers is not in the document]
Comment (davidmallasen) [reply hw/fpga/board_files/vendor/esl_epfl_pynq_z2_board_files.lock.hjson:1]: This is written automatically by the script in `x-heep/util/vendor.py`.

File: hw/fpga/board_files/vendor/esl_epfl_pynq_z2_board_files.vendor.hjson [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Copyright 2023 OpenHW Group
```
[the remark this answers is not in the document]
Comment (davidmallasen) [reply hw/fpga/board_files/vendor/esl_epfl_pynq_z2_board_files.vendor.hjson:1]: True, my bad

**Comment by davidmallasen** [comment]
I updated with your comments. The image was there because it appears when you open the project directly in Vivado, but I guess it is not strictly needed.

**Comment by JoseCalero** [comment]
I think this can be merged @davideschiavone :)

**Comment by JoseCalero** [comment]
@davideschiavone tested w/ fpga using hello w. and freertos apps

**Pull request merged**